interfaces/vault_kv/v0/README.md

```diff
@@ -0,0 +1,62 @@
+# `vault-kv`
+
+## Usage
+
+Some charms require a secure key value store. This relation interface describes the expected behavior of any charm claiming to interact with Vault Key Value stores.
+
+## Direction
+
+```mermaid
+flowchart TD
+    Requirer -- mount_suffix, egress_subnet --> Provider
+    Provider -- vault_url, mount, credentials --> Requirer
+```
+
+## Behavior
+
+Both the Requirer and the Provider need to adhere to criteria to be considered compatible with the interface.
+
+### Provider
+
+Provider expectations
+
+- Is expected to provide the vault url
+- Is expected to provide a key value mount, the mount name shall respect the following pattern: `charm-<requirer app>-<user provided suffix>`
+- Is expected to create an approle restricted to the requiring unit's egress subnet.
+- Is expected to create a Juju secret containing a role-id and role-secret-id for each unit
+- Is expected to provide the Juju secret ID in the relation data.
+
+### Requirer
+
+Requirer expectations
+
+- Is expected to provide a mount suffix
+- Is expected to provide an egress subnet for each unit requiring access to the vault key value store.
+  The unit's egress_subnet shall be used to restrict access to the secret backend.
+
+## Relation Data
+
+[\[Pydantic Schema\]](./schema.py)
+
+#### Example
+
+```yaml
+provider:
+  app:
+    vault_url: http://10.152.183.104:8200
+    mount: charm-barbican-secrets
+    credentials: |
+      {
+        "barbican-0": "secret://4f7cc474-a23d-49a2-8b6e-9835c1e08325/cjk5slcrl3uc767oebp0",
+        "barbican-1": "secret://4f7cc474-a23d-49a2-8b6e-9835c1e08325/cjk5slcrl3uc767oebpg"
```

Comment:
Where are we with unit names in cross-model relation contexts? In the past the providing side of the relation see obfuscated unit names so interfaces that make use of the remote-unit-name to target a response to a specific requiring unit don't work correctly when used cross-model.

We've used a pattern with a nonce is presented from each requiring unit that the providing service uses to target specific information being return to a specific requiring unit.

Reply:
>  In the past the providing side of the relation see obfuscated unit names so interfaces that make use of the remote-unit-name to target a response to a specific requiring unit don't work correctly when used cross-model.

That is indeed still the case.

Reply:
Indeed, the requirer units won't be able to get their credentials from the databag, since the provider app will see some obfuscated name for the unit, therefore, using a `nonce` provided by the unit is a good alternative.


However, I'm currently facing this [bug](https://bugs.launchpad.net/juju/+bug/2033486) when trying to share the secrets over CMR. I'll make the change to the interface when I'll know more about the bug

Reply:
Resolved, units will now have to provide a nonce. This nonce will be used to set the secret in the `credentials` field of the application databag.